internal/backend/azure/azure.go

```diff
@@ -250,27 +202,17 @@ func (be *Backend) openReader(ctx context.Context, h restic.Handle, length int,
 	}
 
 	objName := be.Filename(h)
-	blob := be.container.GetBlobReference(objName)
-
-	start := uint64(offset)
-	var end uint64
-
-	if length > 0 {
-		end = uint64(offset + int64(length) - 1)
-	} else {
-		end = 0
-	}
+	blob := be.container.NewBlockBlobURL(objName)
 
 	be.sem.GetToken()
 
-	rd, err := blob.GetRange(&storage.GetBlobRangeOptions{Range: &storage.BlobRange{Start: start, End: end}})
+	resp, err := blob.Download(context.TODO(), offset, int64(length), storage.BlobAccessConditions{}, false, storage.ClientProvidedKeyOptions{})
```

Comment:
Please use the context `ctx` passed to this method. This also applies to all other usages of `context.TODO()`